compiler/qsc/src/interpret.rs

```diff
@@ -315,6 +318,70 @@ impl Interpreter {
         })
     }
 
+    /// Given a package ID, returns all the global items in the package.
+    /// Note this does not currently include re-exports.
+    pub fn get_global_items(
+        &self,
+        package_id: PackageId,
+    ) -> Vec<(Vec<Rc<str>>, Rc<str>, fir::StoreItemId)> {
+        let mut exported_items = Vec::new();
+        let package = &self
+            .compiler
+            .package_store()
+            .get(map_fir_package_to_hir(package_id))
+            .expect("package should exist in the package store")
+            .package;
+        for global in global::iter_package(Some(map_fir_package_to_hir(package_id)), package) {
+            if let global::Kind::Term(term) = global.kind {
+                let store_item_id = fir::StoreItemId {
+                    package: package_id,
+                    item: fir::LocalItemId::from(usize::from(term.id.item)),
+                };
+                exported_items.push((global.namespace, global.name, store_item_id));
+            }
+        }
+        exported_items
+    }
+
+    /// Get the global items defined in the user source passed into initialization of the interpreter.
+    pub fn get_source_package_global_items(
+        &self,
+    ) -> Vec<(Vec<Rc<str>>, Rc<str>, fir::StoreItemId)> {
+        self.get_global_items(self.source_package)
+    }
+
+    /// Get the global items defined in the open package being interpreted, which will include any items
+    /// defined by calls to `eval_fragments` and the like.
+    pub fn get_open_package_global_items(&self) -> Vec<(Vec<Rc<str>>, Rc<str>, fir::StoreItemId)> {
```

Comment:
Elsewhere in this module we call this the `current` package or just `package` (unfortunate choice but here we are). I'd stick with that naming to avoid introducing another term into the same file. (I realize "open package" means something for the incremental compiler, but that's not the same layer of abstraction as this file, so the names change slightly).

The naming of the various packages at play have been a persistent pain point for me so I'll be picky about consistency here, sorry!

Reply:
Good call, I'll update this to be consistent with the patterns in the file.